test/test_generate_pedigree.py

```diff
@@ -0,0 +1,159 @@
+"""
+methods for testing the sample-metadata API queries
+"""
+
+from copy import deepcopy
+
+from unittest.mock import patch
+
+import json
+import os
+
+from helpers.pedigree_from_sample_metadata import (
+    ext_to_int_sample_map,
+    get_fat_pedigree,
+    get_pedigree_for_project,
+)
+
+
+PWD = os.path.dirname(__file__)
+PROJECT = 'fake-project'
+INPUT = os.path.join(PWD, 'input')
+JSON_PED = os.path.join(INPUT, 'mock_pedigree.json')
+LOOKUP_PED = os.path.join(INPUT, 'mock_sm_lookup.json')
+
+SAMPLE_TO_CPG = {
+    'sam1': ['cpg1'],
+    'sam2': ['cpg2'],
+    'sam3': ['cpg3'],
+}
+DIRTY_PED = [
+    {
+        'family_id': 'FAM1',
+        'individual_id': 'sam1',
+        'paternal_id': 'sam2',
+        'maternal_id': 'sam3',
+        'sex': 1,
+        'affected': 1,
+    },
+    {
+        'family_id': 'FAMX',
+        'individual_id': 'samx',
+        'paternal_id': 'samy',
+        'maternal_id': 'samz',
+        'sex': 1,
+        'affected': 1,
+    },
+]
+
+
+@patch('helpers.pedigree_from_sample_metadata.FamilyApi.get_pedigree')
+def test_get_pedigree_for_project(
+    get_ped_patch,
+):
+    """
+
+    :param get_ped_patch:
+    :return:
+    """
+
+    with open(JSON_PED, 'r', encoding='utf-8') as handle:
+        payload = json.load(handle)
+        get_ped_patch.return_value = payload
+        result = get_pedigree_for_project(project=PROJECT)
+        assert result == payload
+
+
+@patch(
+    'helpers.pedigree_from_sample_metadata.ParticipantApi.'
+    'get_external_participant_id_to_internal_sample_id'
+)
+def test_ext_to_int_sample_map(
+    map_mock,
+):
+    """
+    fetch method using a mocked API endpoint
+    :param map_mock:
+    :return:
+    """
+
+    with open(LOOKUP_PED, 'r', encoding='utf-8') as handle:
+        payload = json.load(handle)
+        map_mock.return_value = payload
+        result = ext_to_int_sample_map(project=PROJECT)
+        assert isinstance(result, dict)
+        assert result == {
+            'FAM1_father': ['CPG11'],
+            'FAM1_mother': ['CPG12'],
+            'FAM1_proband': ['CPG13'],
+            'FAM2_proband': ['CPG41'],
+        }
```

Comment:
It doesn't matter too much, but good to close the file once you're finished with it, then do stuff after file is closed.

```suggestion
    map_mock.return_value = payload
    result = ext_to_int_sample_map(project=PROJECT)
    assert isinstance(result, dict)
    assert result == {
        'FAM1_father': ['CPG11'],
        'FAM1_mother': ['CPG12'],
        'FAM1_proband': ['CPG13'],
        'FAM2_proband': ['CPG41'],
    }
```